packages/component-navigator/src/hooks/useMarginOverlay.tsx

```diff
@@ -0,0 +1,43 @@
+import { getId } from "@atrilabs/core";
+import {
+  addOrModifyHintOverlays,
+  removeHintOverlays,
+} from "@atrilabs/canvas-runtime";
+import { orange600 } from "@atrilabs/design-system";
+
+type FilledLineProps = {
+  fill: string;
+};
+
+const FilledLine: React.FC<FilledLineProps> = (props) => {
+  return (
+    <div
+      style={{ border: `solid ${props.fill} 2px`, height: "100%", width: "100%" }}
+    ></div>
+  );
+};
+
+export const clickOverlay = getId();
+export const hoverOverlay = getId();
+
+export const useMarginOverlay = (overlay: string) => {
```

Comment:
Instead of creating hint overlays yourself, please refer to this https://github.com/Atri-Labs/atrilabs-engine/discussions/392. The benefits of doing this are also explained in the same discussion.